libraries/provider_systemd_unit.rb

```diff
@@ -17,6 +17,17 @@ def whyrun_supported?
     action :create do
       unit_path = Systemd::Helpers.unit_path(new_resource)
 
+      if new_resource.drop_in && new_resource.override.nil?
+        fail(
+          Chef::Exceptions::ValidationFailed,
+          'Required argument override is missing!'
+        )
+      end
```

Comment:
downside of using LWRP instead of HWRP, don't seem to be able to do this in the resource, even with lazy...